### PR TITLE
update Fields to support description and detail props - update inputText to have width 100% to accommodate the changes

### PR DESCRIPTION
**Pull request by laraharrow**
✨ Changes

- update Fields to support description and details
- update inputText to have a width of 100%

✅ Requirements
- [x] Includes test coverage for all changes
- [x] Build and tests are passing
- [x] PR is ideally < 400LOC
- [x] Link(s) to related Github issues

📷 Screenshots

Screen Shot 2020-02-13 at 4 54 36 PM

### CODE REVIEW

**Comment by laraharrow** [comment]
I have not updated the snapshot but when I changed the width on InputText inside field it created some snapshots conflicts.

File: packages/components/src/Form/Fields/FieldText/FieldText.tsx [hunk 1]
```diff
@@ -49,6 +49,7 @@ const FieldTextComponent = forwardRef(
           id={id}
           validationType={validationMessage && validationMessage.type}
           ref={ref}
+          width="100%"
```
Comment (lukelooker) [inline packages/components/src/Form/Fields/FieldText/FieldText.tsx]: This change will have pretty broad effects on existing uses. Was this called out in the designs?

Comment (laraharrow) [reply packages/components/src/Form/Fields/FieldText/FieldText.tsx]: this was something suggested by Jared.

Comment (lukelooker) [reply packages/components/src/Form/Fields/FieldText/FieldText.tsx]: k. I think this change will introduce too much breakage in existing use-cases. 

If we need 100% width `InputText` for `InlineEdit` let's apply it more specifically.

File: packages/components/src/Form/Fields/Field.tsx [hunk 2]
```diff
@@ -66,6 +67,14 @@ export interface FieldProps {
...
    */
   alignValidationMessage?: FormControlDirections
+  /*
+   * optional extra description
+   */
+  description?: ReactNode
+  /**
+   * notes and details added to the top right corner of the field
+   */
+  detail?: ReactNode
```
[the remark this answers is not in the document]
Comment (laraharrow) [reply packages/components/src/Form/Fields/Field.tsx]: good call!

**Pull request merged**